modules/openapi-generator/src/main/resources/JavaSpring/libraries/spring-boot/pom.mustache

```diff
@@ -161,6 +161,13 @@
             <groupId>com.fasterxml.jackson.dataformat</groupId>
             <artifactId>jackson-dataformat-yaml</artifactId>
         </dependency>
+        {{#useOptional}}
+        <dependency>
+           <groupId>com.fasterxml.jackson.datatype</groupId>
+           <artifactId>jackson-datatype-jdk8</artifactId>
+           <version>2.9.6</version>
+        </dependency>
+        {{/useOptional}}
```

Comment:
About the maven plug-in test failure, I think you will also need to update modules/openapi-generator-maven-plugin/examples/spring.xml

Test: https://github.com/OpenAPITools/openapi-generator/blob/master/.github/workflows/openapi-generator.yaml#L188